olp-cpp-sdk-dataservice-read/src/generated/api/StreamApi.cpp

```diff
@@ -0,0 +1,134 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#include "StreamApi.h"
+
+#include <map>
+
+#include <olp/core/client/CancellationContext.h>
+#include <olp/core/client/HttpResponse.h>
+#include <olp/core/client/OlpClient.h>
+#include <olp/core/generated/parser/JsonParser.h>
+#include <olp/core/logging/Log.h>
+#include <olp/dataservice/read/model/Data.h>
+#include "generated/parser/SubscribeResponseParser.h"
+#include "generated/serializer/ConsumerPropertiesSerializer.h"
+#include "generated/serializer/JsonSerializer.h"
+
+namespace {
+constexpr auto kLogTag = "read::StreamApi";
+}  // namespace
+
+namespace olp {
+namespace dataservice {
+namespace read {
+
+StreamApi::SubscribeApiResponse StreamApi::Subscribe(
+    const client::OlpClient& client, const std::string& layer_id,
+    const boost::optional<std::string>& subscription_id,
+    const boost::optional<std::string>& mode,
+    const boost::optional<std::string>& consumer_id,
+    const boost::optional<ConsumerProperties>& subscription_properties,
+    const client::CancellationContext& context,
+    std::string& x_correlation_id /*out*/) {
+  std::string metadata_uri = "/layers/" + layer_id + "/subscribe";
+
+  std::multimap<std::string, std::string> query_params;
+  if (subscription_id) {
+    query_params.insert(
+        std::make_pair("subscriptionId", subscription_id.get()));
+  }
+
+  if (mode) {
+    query_params.insert(std::make_pair("mode", mode.get()));
+  }
+
+  if (consumer_id) {
+    query_params.insert(std::make_pair("consumerId", consumer_id.get()));
+  }
+
+  std::multimap<std::string, std::string> header_params;
+  header_params.insert(std::make_pair("Accept", "application/json"));
+
+  model::Data data;
+  if (subscription_properties) {
+    auto serialized_subscription_properties =
+        serializer::serialize(subscription_properties.get());
+    data = std::make_shared<std::vector<unsigned char>>(
+        serialized_subscription_properties.begin(),
+        serialized_subscription_properties.end());
+  }
+
+  client::HttpResponse response = client.CallApi(
+      metadata_uri, "POST", std::move(query_params), std::move(header_params),
+      {}, data, "application/json", std::move(context));
+  if (response.status != http::HttpStatusCode::CREATED) {
+    return client::ApiError(response.status, response.response.str());
+  }
+
+  OLP_SDK_LOG_TRACE_F(kLogTag, "subscribe, uri=%s, status=%d",
```

Comment:
```suggestion
  OLP_SDK_LOG_DEBUG_F(kLogTag, "subscribe, uri=%s, status=%d",
```
Trace is usually only for method entry/exit.